clojure-ts-mode.el

```diff
@@ -2032,6 +2055,135 @@ value is `clojure-ts-thread-all-but-last'."
+         (same-line-p (= line-parent line-args))
+         (single-arity-p (clojure-ts--vec-node-p first-coll)))
+    (goto-char coll-start)
+    (when same-line-p
+      (newline-and-indent))
+    (when single-arity-p
+      (insert-pair 2 ?\( ?\))
+      (backward-up-list))
+    (insert "([])\n")
+    ;; Put the point between square brackets.
+    (down-list -2)))
+
+(defun clojure-ts--add-arity-defprotocol-internal (fn-node)
+  "Add an arity to a defprotocol function defined by FN-NODE."
+  (let* ((args-vec (clojure-ts--node-child fn-node (rx bol "vec_lit" eol)))
+         (args-vec-start (clojure-ts--node-start-skip-metadata args-vec))
+         (line-parent (thread-first fn-node
+                                    (clojure-ts--node-child-skip-metadata 0)
+                                    (treesit-node-start)
+                                    (line-number-at-pos)))
+         (line-args-vec (line-number-at-pos args-vec-start))
+         (same-line-p (= line-parent line-args-vec)))
+    (goto-char args-vec-start)
+    (insert "[]")
+    (if same-line-p
+        (insert " ")
+      ;; If args vector is not at the same line, respect this and place each new
+      ;; vector on a new line.
+      (newline-and-indent))
+    ;; Put the point between square brackets.
+    (down-list -1)))
+
+(defun clojure-ts--add-arity-reify-internal (fn-node)
+  "Add an arity to a reify function defined by FN-NODE."
+  (let* ((fn-name (clojure-ts--list-node-sym-text fn-node)))
+    (goto-char (clojure-ts--node-start-skip-metadata fn-node))
+    (insert "(" fn-name " [])")
+    (newline-and-indent)
+    ;; Put the point between sqare brackets.
+    (down-list -2)))
+
+(defun clojure-ts--letfn-defn-p (node)
+  "Return non-nil if NODE is a function definition in a letfn form."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (clojure-ts--vec-node-p parent)
+         (let ((grandparent (treesit-node-parent parent)))
+           (string= (clojure-ts--list-node-sym-text grandparent)
+                    "letfn")))))
+
+(defun clojure-ts--proxy-defn-p (node)
+  "Return non-nil if NODE is a function definition in a proxy form."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (string= (clojure-ts--list-node-sym-text parent) "proxy"))))
+
+(defun clojure-ts--defprotocol-defn-p (node)
+  "Return non-nil if NODE is a function definition in a defprotocol form."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (string= (clojure-ts--list-node-sym-text parent) "defprotocol"))))
+
+(defun clojure-ts--reify-defn-p (node)
+  "Return non-nil if NODE is a function definition in a reify form."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (string= (clojure-ts--list-node-sym-text parent) "reify"))))
+
+(defun clojure-ts-add-arity ()
+  "Add an arity to a function."
```

Comment:
I think the docstring should be extended to reflect that this actually with other function-like constructs.

Reply:
Changed to `Add an aarity to a function or macro`.